hyper/http20/connection.py

```diff
@@ -130,18 +179,26 @@ def __init_state(self):
         users should be strongly discouraged from messing about with connection
         objects themselves.
         """
-        self._conn = h2.connection.H2Connection()
+        self._conn = _LockedH2Connection(h2.connection.H2Connection(),
+                                         self._lock)
```

Comment:
It feels a little bit weird to use the general lock for this: should we have a lock internal to the locked connection that just serializes there? I'm worried about unnecessary lock contention if we re-use this.